File: problemsets/Codeforces/Python/A1005.py
"""
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 """
input()
a=[*map(int,input().split())]
m=[x for x,y in zip(a,a[1:]+[1]) if y==1]
print(len(m))
print(*m)

File: problemsets/Codeforces/Go/A1005.go
/**
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 */
package main
import "fmt"
func main() {
	var n, x int
	l := 0
	r := []int{}
	fmt.Scan(&n)
	for i:=0; i < n; i++ {
		fmt.Scan(&x)
		if x <= l {
			r = append(r, l)
		}
		l = x
	}
	r = append(r, l)
	fmt.Println(len(r))
	for _,x := range r {
		fmt.Printf("%d ", x)
	}
}